server/session.py

```diff
@@ -552,10 +553,16 @@ def notify(self, height, touched):
         '''Notify the client about changes in masternode list.'''
         result = super().notify(height, touched)
 
-        for masternode in self.mns:
-            status = self.daemon.masternode_list(['status', masternode])
-            self.send_notification('masternode.subscribe',
-                                   [masternode, status.get(masternode)])
+        try:
+            for masternode in self.mns:
+                status = self.daemon.masternode_list(['status', masternode])
+                if hasattr(status, 'get'):
+                    self.send_notification('masternode.subscribe',
+                                           [masternode, status.get(masternode)])
+        except DaemonError as e:
+            error, = e.args
+            message = error['message']
+            self.logger.info('there was an error when notifying the masternode status: {}'.format(message))
```

Comment:
I added this try/catch block because sometimes I got this error:

```
electrumx_server.py[13104]: ERROR:Controller:Traceback (most recent call last):
electrumx_server.py[13104]:   File "/usr/local/lib/python3.6/dist-packages/electrumx-1.3.1-py3.6.egg/server/controller.py", line 178, in on_future_done
electrumx_server.py[13104]:     future.result()
electrumx_server.py[13104]:   File "/usr/local/lib/python3.6/dist-packages/electrumx-1.3.1-py3.6.egg/server/block_processor.py", line 206, in main_loop
electrumx_server.py[13104]:     await task()
electrumx_server.py[13104]:   File "/usr/local/lib/python3.6/dist-packages/electrumx-1.3.1-py3.6.egg/server/block_processor.py", line 256, in check_and_advance_blocks
electrumx_server.py[13104]:     self.controller.mempool.on_new_block(self.touched)
electrumx_server.py[13104]:   File "/usr/local/lib/python3.6/dist-packages/electrumx-1.3.1-py3.6.egg/server/mempool.py", line 194, in on_new_block
electrumx_server.py[13104]:     self.controller.notify_sessions(touched)
electrumx_server.py[13104]:   File "/usr/local/lib/python3.6/dist-packages/electrumx-1.3.1-py3.6.egg/server/controller.py", line 296, in notify_sessions
electrumx_server.py[13104]:     session_touched = session.notify(height, touched)
electrumx_server.py[13104]:   File "/usr/local/lib/python3.6/dist-packages/electrumx-1.3.1-py3.6.egg/server/session.py", line 608, in notify
electrumx_server.py[13104]:     'result': status.get(masternode),
electrumx_server.py[13104]: AttributeError: 'coroutine' object has no attribute 'get'
```
I know it's not the best way but it's just a workaround, I'll try to fix it later.

Reply:
You're right, it's not a workaround at all.  You will get that every time.  The clue is in the error message - it's a coroutine because you missed the "await".

Reply:
yeap, I tried by adding the 'await' but I got 'syntax error' because the 'notify' method is not async, also I tried using the asyncio library to wait until the thread is done but I got another error

Reply:
I see.  Check out the notify_async function - which I created for async notification things.  Ultimately this will all be async, but for now see if you can modify the way that works to achieve what you want.

Reply:
question, at what moment notify_async is triggered? because I'm figuring out how it works

Reply:
I already tested and that code can be implemented in `notify_async `or in `notify `either using the `controller.create_task `method.

That's why I'm asking about `notify_async `